QHttp and QUrlQuery

maurerpe opened the pull request:
I completely removed all QHttp code and replaced it with QNetwork\* code which is valid for both Qt4 and Qt5.  This removes the need to for 3rdParty QtHttp.  I also filled in some fixme_qt5 blocks using QUrlQuery.

f3nix committed:
Add CMake option to build FreeCAD with Qt5.
Based on previous work of Peter Lama (peterl94).

f3nix committed:
qInstallMsgHandler() is obsolete in Qt5. Use qInstallMessageHandler() instead.

This change is Qt5 specific.

f3nix committed:
QCoreApplication::UnicodeUTF8 is obsolete in Qt5. Remove all instances.

This change is Qt5 specific.

f3nix committed:
QApplication::UnicodeUTF8 is obsolete in Qt5. Remove all instances.

This change is Qt5 specific.

f3nix committed:
QCoreApplication::CodecForTr is obsolete in Qt5. Remove all instances.

This change is Qt5 specific.

f3nix committed:
Make Base::Quantity known to QMetaType.

This change is Qt4/Qt5 neutral.

f3nix committed:
QHeaderView::setResizeMode() is obsolete in Qt5. Use setSectionResizeMode() instead.

This change is Qt5 specific.

f3nix committed:
QHeaderView::setMovable() is obsolete in Qt5. Use setSectionsMovable() instead.

This change is Qt5 specific.

f3nix committed:
Qt::escape() is obsolete in Qt5. Use QString::toHtmlEscaped() instead.

This change is Qt5 specific.

f3nix committed:
QDesktopServices::storageLocation() is obsolete in Qt5. Use QStandardPaths::standardLocations() instead.
Use enums from QStandardPaths instead of QDesktopServices.

This change is Qt5 specific.

f3nix committed:
QFile::setDecodingFunction() and QFile::setEncodingFunction() are obsolete in Qt5. Do not use them.

This change is Qt5 specific.

f3nix committed:
QHeaderView::setClickable() is obsolete in Qt5. Use setSectionsClickable() instead.

This change is Qt5 specific.

f3nix committed:
Remove virtual keyword from PropertyItem to allow compilation with Qt5. :toilet:

f3nix committed:
Add preliminary support for building with PySide2.

This change is Qt5 specific.

j-wiedemann committed:
Sketcher : Allow µm in sketcher grid size

maurerpe committed:
Add missing header files

Needed for Qt5, recommended for Qt4

maurerpe committed:
Ensure mutexes are always properly unlocked

Error exists with Qt4 and Qt5.  Qt5 generated helpful warnings that made
it easier to debug.

maurerpe committed:
Change numBytes() to byteCount()

numBytes() is obsolete in Qt 4.8 and removed in Qt5.  byteCount() is a
drop in replacement that works in 4.8 and 5.

maurerpe committed:
Fix remaining toAscii/fromAscii

Qt5 does not have toAscii or fromAscii.  Use toLatin1 and fromLatin1
respectively.  Both replacement functions exist in Qt4.8.

yorikvanhavre committed:
Merge pull request #101 from wood-galaxy/sketcher-micro-sizing

Sketcher : Allow µm in sketcher grid size

yorikvanhavre committed:
Made Redirect python output options True by default - fixes #2454

yorikvanhavre committed:
Turned Zoom At Cursor option True by default - fixes #2371

yorikvanhavre committed:
Draft: Fixed undo mode of wires - fixes #2456

yorikvanhavre committed:
Added updated translations from crowdin - fixes #2443

triplus committed:
Add SVG workbench icons

triplus committed:
Create RadiosityOutdoorHQ.pov

triplus committed:
Added RadiosityOutdoorHQ.pov

triplus committed:
Added RadiosityOutdoorHQ.pov

wwmayer committed:
+ fix typo in Raytracing dialog

wwmayer committed:
Merge branch 'master' of https://github.com/FreeCAD/FreeCAD

wwmayer committed:
+ use quantity box in image orientation dialog

wwmayer committed:
+ use quantity box in tessellation dialog

wwmayer committed:
+ add convenience methods to PrefQuantitySpinBox class

wwmayer committed:
+ use quantity box in DlgSettingsImportExport

yorikvanhavre committed:
Merge pull request #103 from triplus/RadiosityOutdoorHQ

Add RadiosityOutdoorHQ.pov Raytracing Template

yorikvanhavre committed:
Draft: Fixed placement bug in clones

wwmayer committed:
+ use quantity box for length on extrusion dialog

wwmayer committed:
+ use quantity boxes in inspection dialog

wwmayer committed:
+ use quantity boxes in cross section dialog

wwmayer committed:
+ use quantity box in offset dialog

yorikvanhavre committed:
Updated Mac icon with the new style

yorikvanhavre committed:
Merge pull request #102 from triplus/SVG_WB_Icons

Add SVG workbench icons

wwmayer committed:
+ use quantity boxes in revolution dialog

wwmayer committed:
+ use quantity boxes in box & cylinder creation dialogs

wwmayer committed:
+ add quantity boxes to mirroring dialog

ianrrees committed:
Resolve clang compile error

yorikvanhavre committed:
Merge pull request #104 from ianrrees/20160228-fix-mac-builds

Resolve clang compile error

wwmayer committed:
+ use quantity boxes in fillet/chamfer dialog

wwmayer committed:
+ make fillet/chamfer dialog narrower

wwmayer committed:
+ use quantity boxes in location widget

wwmayer committed:
+ fixes #0002049: 0.15 and 0.16 Part Revolve ignores unit accuracy setting beyond 2 decimals

wwmayer committed:
+ use application-wide number of decimals in primitive creation dialog

wwmayer committed:
+ use quantity boxes in mesh solid creation dialog

wwmayer committed:
+ use quantity boxes in mesh segmentation dialog

yorikvanhavre committed:
Arch: Fixed bug in extrusion of structures - fixes #2458

wwmayer committed:
+ fixes #0002200: Application crash when modifying fillet

wwmayer committed:
+ code clean-up

yorikvanhavre committed:
Draft: better docstring for offset - fixes #2199

bblacey committed:
Added High Resolution Display icons to Mac OS X iconset resource file (.icns).

yorikvanhavre committed:
Start: Updated startpage - fixed some links + restored commits feed

yorikvanhavre committed:
Merge pull request #106 from bblacey/update-osx-0.16-iconset

Added High Resolution Display icons to Mac OS X iconset resource file…

maurerpe committed:
Merge branch 'master' of https://github.com/FreeCAD/FreeCAD

maurerpe committed:
Merge branch 'qt5-transition-5' of https://github.com/f3nix/FreeCAD into qt5

maurerpe committed:
Use QNetwork* instead of obsolute QHttp.

QNetwork* is available in Qt 4.8 so this change is Qt4/Qt5 neutral.

maurerpe committed:
Cleaner fix for Base::Quantity

All that was needed was to add include PropertyItem.h in
DlgUnitsCalculatorImp.cpp.  This keeps all the Q_DECLARE_METATYPES in
one place.

ovginkel committed:
FEM: remove module ccxInpWriterFemConstraintDisplacement

which was intoduced in commit b1dc2cb by an accident

maurerpe committed:
Use QUrlQuery to fix url query issues

Change is Qt5 specific.

maurerpe committed:
Update for PyQt5

ovginkel committed:
FEM: remove some trailing whitespaces

ovginkel committed:
FEM: do not hide constraints for result view

ovginkel committed:
FEM: C++98 compatible fix to constraint displacement reference deletion, update vertice, edge, face selection

wwmayer committed:
+ set range for quantity boxes

wwmayer committed:
+ prepare string for translation

maurerpe committed:
Use QUrlQuery to fix url query issues
Change is Qt5 specific.

wwmayer committed:
+ change binary export & import of shapes to restore placement and orientation

maurerpe committed:
Use QNetwork* instead of obsolete QHttp.
QNetwork* is available in Qt 4.8 so this change is Qt4/Qt5 neutral.

wwmayer committed:
+ fix build failure with gcc

f3nix commented:
Hi @maurerpe!
Thanks for the PR.

I have created a new rebased branch https://github.com/f3nix/FreeCAD/tree/qt5-transition-6

I have cherry picked two of your commits:
https://github.com/maurerpe/FreeCAD/commit/6cc526e14549d256cb5a18e1166dc53b4a7bffc8 with some white characters changes + build failure fix

https://github.com/maurerpe/FreeCAD/commit/5824b6b871add6a58d2842085d552b88879bfed5 with some white characters changes

Please see comments on the other two.

Could You please make future PR's rebased on the latest qt5-transition branch? It would be very welcome.

Cheers,
Mateusz

maurerpe committed:
Fix build error caused by mispelling quantity

Quantity was misspelled as quanity.

maurerpe committed:
Fix PyQt5 loadUi

Existing python scripts expect loadUi to define an attribute for each
named child.  PySide does this automatically, PyQt5 does not.

maurerpe committed:
Merge branch 'qt5-transition-6' of https://github.com/f3nix/FreeCAD into qt5

maurerpe committed:
Make CMakelists.txt match f3nix's version

maurerpe committed:
Revert Quantity.h and PropertyItem.h

Make Quantity.h and PropertyItem.h match FreeCAD/FreeCAD more closely.
This is allowed by including PropertyItem.h in DlgUnitsCalculatorImp.cpp
